src/charm.py

```diff
@@ -147,6 +147,20 @@ def _additional_hostnames(self) -> Generator[str, None, None]:
         additional_hostnames = self._get_config_or_relation_data("additional-hostnames", "")
         yield from filter(None, additional_hostnames.split(","))
 
+    @property
+    def _backend_protocol(self) -> str:
+        """Return the backend-protocol to use for k8s ingress."""
+        backend_protocol_default = "HTTP"
+        backend_protocol = self._get_config_or_relation_data(
+            "backend-protocol", backend_protocol_default
+        ).upper()
+        # Disabled to reference the documentation
+        # See https://kubernetes.github.io/ingress-nginx/user-guide/nginx-configuration/annotations/#backend-protocol # noqa: E501 pylint: disable=line-too-long
+        accepted_values = ("HTTP", "HTTPS", "GRPC", "GRPCS", "AJP", "FCGI")
+        if backend_protocol not in accepted_values:
```

Comment:
Perhaps it would be a better UX to block the charm if the value is invalid

Reply:
I was in doubt about that because "retry-errors" configuration option does not block so I followed the pattern.
From the config.yaml : "Unrecognised values will be ignored. The nginx default will be used if this config option is set to an empty value."
I could block or updated the doc warning about this behavior as retry-errors does. WDYT?